### ISSUE TITLE
Cell colors and RichText colors are incorrect when theme colors are used

### ISSUE THREAD

**Issue by michaelschroeder2** [EPPlusSoftware/EPPlus#743]
- Issue 1: LookUp of cell color of theme or automatic returns wrong colors, though getting them directly from theme gets the correct values
- Issue 2: theme names in excel differ from names in lib (text1/2 vs dark1/2, background1/2 vs light1/2) , though everyone who knows that, can do a manual mapping
- Issue 3: returned color of RichText is always 0 if theme is used ... nothing can be done

used excel file:
[Colors.xlsx](https://github.com/EPPlusSoftware/EPPlus/files/9938516/Colors.xlsx)

Test code:
```
using OfficeOpenXml;
using OfficeOpenXml.Drawing.Style.Coloring;
using System;
using System.IO;

namespace GetColors
{
	class Program
	{
		static void Main(string[] args)
		{
			ExcelPackage.LicenseContext = LicenseContext.NonCommercial;

			var userName = "I";
			var fileInfo = new FileInfo($@"C:\Users\{userName}\Desktop\Colors.xlsx");

			using (var excelPackage = new ExcelPackage(fileInfo))
			{
				int[] rows = { 1, 3, 5, 7, 9, 12, 14, 16, 18, 20 };

				foreach (var row in rows)
				{
					var cell = excelPackage.Workbook.Worksheets[0].Cells[row, 3];
					var cellColor = "";

					if (!String.IsNullOrEmpty(cell.Style.Font.Color.Theme.ToString()))
					{
						var theme = cell.Style.Font.Color.Theme.ToString();

						if (theme == "Text1") theme = "Dark1";
						else if (theme == "Text2") theme = "Dark2";
						else if (theme == "Background1") theme = "Light1";
						else if (theme == "Background2") theme = "Light2";

						var colorManager = (ExcelDrawingThemeColorManager)excelPackage.Workbook.ThemeManager.CurrentTheme.ColorScheme
														.GetType()
														.GetProperty(theme)
														.GetValue(excelPackage.Workbook.ThemeManager.CurrentTheme.ColorScheme);

						switch (colorManager.ColorType)
						{
							case OfficeOpenXml.Drawing.eDrawingColorType.None:
								break;
							case OfficeOpenXml.Drawing.eDrawingColorType.RgbPercentage:
								break;
							case OfficeOpenXml.Drawing.eDrawingColorType.Rgb:
								cellColor = '#' + colorManager.RgbColor.Color.Name.ToUpper();
								break;
							case OfficeOpenXml.Drawing.eDrawingColorType.Hsl:
								break;
							case OfficeOpenXml.Drawing.eDrawingColorType.System:
								cellColor = '#' + colorManager.SystemColor.LastColor.Name.ToUpper();
								break;
							case OfficeOpenXml.Drawing.eDrawingColorType.Scheme:
								break;
							case OfficeOpenXml.Drawing.eDrawingColorType.Preset:
								break;
							case OfficeOpenXml.Drawing.eDrawingColorType.ChartStyleColor:
								break;
							default:
								break;
						}
					}

					Console.WriteLine(excelPackage.Workbook.Worksheets[0].Cells[row, 1].Text);
					Console.WriteLine("Cell RGB: " + cell.Style.Font.Color.Rgb);
					Console.WriteLine("Cell LookUp: " + cell.Style.Font.Color.LookupColor());
					Console.WriteLine("Cell Theme: " + cellColor);

					if (cell.RichText != null)
					{
						foreach (var richText in cell.RichText)
						{
							Console.WriteLine(richText.Text + " " + richText.Color.Name.ToUpper());
						}
					}
					Console.WriteLine();
				}

				Console.ReadKey();
			}
		}
	}
}
```

Screenshots:
![image](https://user-images.githubusercontent.com/50169678/199981268-7fa39818-3ae8-4083-aca4-65cd1b5da33c.png)

![image](https://user-images.githubusercontent.com/50169678/199982172-b76347aa-7635-4531-add8-e8be8a7628f3.png)

**Comment by JanKallman** [comment]
I will have a look

**Comment by JanKallman** [comment]
1. Yes, LookupColor currently don't look at the Automatic color or the Theme color. I'll see if I can add a fix.
2. This is the OOXML names used. They are for some reason different than Excel's. 
3. Yes, we should add a new property with support for app properties - Theme, Tint and Shade.

**Comment by JanKallman** [comment]
Try fix in latest develop build ... https://github.com/EPPlusSoftware/EPPlus/wiki/Using-our-Develop-Nuget-Feed

**Comment by michaelschroeder2** [comment]
Thank you for looking into this topic that fast!

Note: I have updated the code in the original post to show more details. The attached excel file is still the same.
With that you can do tests easily yourself.

The question in the background is, what should LookUpColor() return?

### Part1: CellFontColors:
Currently the methode LookUpColor on cells with theme colors returns the **BaseColor** of the theme.
Along with the tint of the cell (cell.Style.Font.Color.Tint) the final color can then be calculated.

Is this intended? Or should LookUpColor() return the actual color that the user sees in the excel file?

![image](https://user-images.githubusercontent.com/50169678/200812410-8fff237d-a3a7-4e4f-a280-b7bc70deb1ef.png)


### Part2: RichTextColor with theme color derived from cell
RichText does not have a tint or theme or anything else, just color. So it should always show the final color as seen.
So the colors of AAAAA in the sample are wrong since they show just the base color.

And to hold things consistent, I personally think that cell LookUpColor() also should return the color as seen and not the Base Color.

In the sample .xlsx file the RichText cells themselves got theme Accent2 with +40%, then the letters BBBBB have been selected and got a different colors themselves.

![image](https://user-images.githubusercontent.com/50169678/200819142-1bb889fa-a716-4ca7-a2ab-dc4eb2069076.png)

![image](https://user-images.githubusercontent.com/50169678/200821470-754e8547-0479-441f-87c7-c920293bd45a.png)


### Part3: RichTextColor individually set
The colors of the individual richtext colors seem to be fine, with one exception: Automatic(black)

![image](https://user-images.githubusercontent.com/50169678/200821385-01abdfd9-c9cd-4647-aac6-83d39609be5b.png)

**Comment by JanKallman** [comment]
Hello again,
With the change I made now, the intention is that the color should be the same as the color of the cell in Excel. I'll have another look at this and will try to correct the issues you point out. Thanks.

**Comment by JanKallman** [comment]
Fixes added in EPPlus 6.1.1
